### PR TITLE
Ready for review

### PR DESCRIPTION
**Pull request by idontgetoutmuch**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ bumped version for float/double range bugfix
 2. Support for monadic generators e.g. [mwc-random](https://hackage.haskell.org/package/mwc-random).
 3. Monadic adapters for pure generators (providing a uniform monadic
    interface to pure and monadic generators).
-4. Faster by more x10 (depending on the type) - see below for benchmarks.
+4. Faster by more x1000 (depending on the type) - see below for benchmarks.
```
Comment (curiousleo) [inline CHANGELOG.md]: Super-nit: I tend to be skeptical of claims like "better by up to <number>". In this case, even the _least_ improved generator is impressive, so I think we should mention it. i.e. "40 to 1000 times faster pseudo-random value generation". (I think 40 is about right, but worth checking again).

File: CHANGELOG.md [hunk 2]
```diff
@@ -49,46 +49,47 @@ bumped version for float/double range bugfix
...
-	| Chars        |            790 |            133 |                  5.9 |
-	| BIG Integers |         199848 |         103056 |                  1.9 |
-	|--------------|----------------|----------------|----------------------|
-
-
-
+Here are some benchmarks run on a 3.1 GHz Intel Core i7. The full
+benchmarks can be run using e.g. `stack bench`. The benchmarks are
+measured in milliseconds per 100,000 generations. In some cases, the
+performance is over x1000 times better.
```
Comment (curiousleo) [inline CHANGELOG.md]: Same here - I give much more weight to such claims personally when a lower bound is also given.

**Review by curiousleo — COMMENTED** [review]
Thank you for doing this! I find claims of performance improvements much more credible when a lower bound is given too. In this case the lower bound is really respectable, so let's include it!

File: CHANGELOG.md (hunk 1 above)
Comment (idontgetoutmuch) [reply CHANGELOG.md]: Well the minimum is INTEGER which is only 13% faster - the next lowest is x18 for types with 8 bits - I think if this is too heavily caveated you lose the message

Comment (curiousleo) [reply CHANGELOG.md]: Fair enough. No need to dwell on this too much. The other alternative that comes to mind would be e.g. "Faster generation for all implemented types, by up to x1000 for fixed-width integral types".

But if you think that's too much caveating, I'm fine with keeping this as it is.

Comment (lehins) [reply CHANGELOG.md]: Maybe something vague, but still accurate with a lower bound, along the lines: "Performance improvements all across the board with generation of most types being faster on the orders of x100 to x1000"

File: CHANGELOG.md [hunk 3]
```diff
@@ -30,7 +30,9 @@ bumped version for float/double range bugfix
 2. Support for monadic generators e.g. [mwc-random](https://hackage.haskell.org/package/mwc-random).
 3. Monadic adapters for pure generators (providing a uniform monadic
    interface to pure and monadic generators).
-4. Faster by more x1000 (depending on the type) - see below for benchmarks.
+4. Faster in all cases bar one by more than x18 (N.B. x18 not 18%) and
```
Comment (lehins) [inline CHANGELOG.md]: What does this mean: `bar one by`? Was that intended or am I missing a piece of English vocabulary?

**Review by lehins — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (idontgetoutmuch) [reply CHANGELOG.md]: bar means except probably more common in the North - the exception is Integer - I will replace bar by except

There's a famous song: On Ilkley Moor bar t'hat (on Ilkley Moor without a hat) - actually it's probably not that famous - my children wouldn't know it.

**Comment by curiousleo** [comment]
Looking at the whole file, I just noticed this: I've seen a bunch of projects adding new releases to the _top_ of the changelog rather than at the bottom. I guess this way the most recent = relevant changes are visible immediately. The most recent previous release, 1.1, is at the top. In the interest of consistency, I suggest moving the 1.2 release notes to the beginning of the file.

**Comment by idontgetoutmuch** [comment]
> Looking at the whole file, I just noticed this: I've seen a bunch of projects adding new releases to the _top_ of the changelog rather than at the bottom. I guess this way the most recent = relevant changes are visible immediately. The most recent previous release, 1.1, is at the top. In the interest of consistency, I suggest moving the 1.2 release notes to the beginning of the file.

Excellent idea!

**Review by curiousleo — APPROVED** [review]
LGTM!

**Pull request merged**